config/opal_get_version.m4

```diff
@@ -80,7 +80,7 @@ m4_define([OPAL_GET_VERSION],[
 
             # If we're in a git repo and we found the git command, use
             # git describe to get the repo rev
-            if test -d "$srcdir/.git" && test $git_happy -eq 1; then
+            if test -r "$srcdir/.git" && test $git_happy -eq 1; then
```

Comment:
Why wouldn't `.git` always be a directory (if it exists)?

Reply:
This is how `git worktree` is implemented. (basically, a single `.git` directory is shared between many "checkout'ed" branches.
For example, I `git clone` in `Users/gilles/src/ompi-master`, and then in that directory `git worktree add ../ompi-v4.1.x v4.1.x`.

```
$ ls -l /Users/gilles/src/ompi-v4.1.x/.git
-rw-r--r--  1 gilles  staff  65 Jun 16 11:59 /Users/gilles/src/ompi-v4.1.x/.git
$ cat /Users/gilles/src/ompi-v4.1.x/.git
gitdir: /Users/gilles/src/ompi-master/.git/worktrees/ompi-v4.1.x
```

Reply:
Gotcha.